selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/stack/CreateWorkspaceFromSpringBootStackTest.java

```diff
@@ -0,0 +1,119 @@
+import org.eclipse.che.selenium.pageobject.Consoles;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.eclipse.che.selenium.pageobject.dashboard.CreateWorkspaceHelper;
+import org.eclipse.che.selenium.pageobject.dashboard.Dashboard;
+import org.openqa.selenium.By;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/** @author Skoryk Serhii */
+public class CreateWorkspaceFromSpringBootStackTest {
+
+  private static final String WORKSPACE_NAME = generate("workspace", 4);
+  private static final String SPRING_BOOT_HEALTH_CHECK_BOOSTER = "spring-boot-health-check-booster";
+  private static final String SPRING_BOOT_HTTP_BOOSTER = "spring-boot-http-booster";
+
+  private List<String> projects =
+      ImmutableList.of(SPRING_BOOT_HEALTH_CHECK_BOOSTER, SPRING_BOOT_HTTP_BOOSTER);
+
+  @Inject private Ide ide;
+  @Inject private Consoles consoles;
+  @Inject private Dashboard dashboard;
+  @Inject private DefaultTestUser defaultTestUser;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private CreateWorkspaceHelper createWorkspaceHelper;
+  @Inject private TestWorkspaceServiceClient workspaceServiceClient;
+
+  @BeforeClass
+  public void setUp() {
+    dashboard.open();
+  }
+
+  @AfterClass
+  public void tearDown() throws Exception {
+    workspaceServiceClient.delete(WORKSPACE_NAME, defaultTestUser.getName());
+  }
+
+  @Test
+  public void checkWorkspaceCreationFromSpringBootStack() {
+    createWorkspaceHelper.createWorkspaceFromStackWithProjects(
+        SPRING_BOOT, WORKSPACE_NAME, projects);
+
+    ide.switchToIdeAndWaitWorkspaceIsReadyToUse(PREPARING_WS_TIMEOUT_SEC * 2);
+
+    projectExplorer.waitProjectInitialization(SPRING_BOOT_HEALTH_CHECK_BOOSTER);
+    projectExplorer.waitProjectInitialization(SPRING_BOOT_HTTP_BOOSTER);
+  }
+
+  @Test(priority = 1)
+  public void checkSpringBootHealthCheckBoosterProjectCommands() {
+    consoles.executeCommandFromProjectExplorer(
+        SPRING_BOOT_HEALTH_CHECK_BOOSTER, BUILD_GOAL, BUILD_COMMAND, BUILD_SUCCESS);
+
+    consoles.executeCommandFromProjectExplorer(
+        SPRING_BOOT_HEALTH_CHECK_BOOSTER, BUILD_GOAL, CLEAN_BUILD_COMMAND, BUILD_SUCCESS);
+
+    consoles.executeCommandFromProjectExplorer(
+        SPRING_BOOT_HEALTH_CHECK_BOOSTER, RUN_GOAL, RUN_COMMAND, "Started BoosterApplication in");
+    consoles.checkWebElementVisibilityAtPreviewPage(
+        By.xpath("//h2[text()='Health Check Booster']"));
+    consoles.closeProcessTabWithAskDialog(RUN_COMMAND);
+
+    consoles.executeCommandFromProcessesArea(
+        "dev-machine", DEBUG_GOAL, DEBUG_COMMAND, LISTENING_AT_ADDRESS);
+    consoles.closeProcessTabWithAskDialog(DEBUG_COMMAND);
+  }
+
+  @Test(priority = 1)
+  public void checkSpringBooHttpBoosterProjectCommands() {
+    consoles.executeCommandFromProjectExplorer(
```

Comment:
It's not quite clear what next sequence of command is doing exactly. Could you, please, comment it in the code?